nuimo/src/main/kotlin/com/senic/nuimo/NuimoBluetoothController.kt

```diff
@@ -125,11 +136,16 @@ class NuimoBluetoothController(bluetoothDevice: BluetoothDevice, context: Contex
         }
 
         override fun onCharacteristicRead(gatt: BluetoothGatt, characteristic: BluetoothGattCharacteristic, status: Int) {
+            writeQueue.next()
+
             if (characteristic.uuid.equals(BATTERY_CHARACTERISTIC_UUID)) {
-                matrixWriter?.onWrite()
                 val batteryPercentage = characteristic.getIntValue(BluetoothGattCharacteristic.FORMAT_UINT8, 0) ?: -1
                 notifyListeners { it.onBatteryPercentageChange(batteryPercentage) }
             }
+            else if (characteristic.uuid.equals(FIRMWARE_VERSION_CHARACTERISTIC_UUID)) {
+                val firmwareVersion = characteristic.getStringValue(0)
```

Comment:
Characteristic is present != Android returns a non-null value. Please test for null, never force un-wrap an optional unless there's a good reason.